arches/app/models/graph.py

```diff
@@ -2523,6 +2524,15 @@ def restore_state_from_serialized_graph(self, serialized_graph):
             if draft_graph:
                 draft_graph.delete()
 
+            views = models.SpatialView.objects.filter(geometrynode__graph=self.pk)
```

Comment:
In general, I'm pushing for us to keep things lazy by using `self` to do queries, for the reason that the nodes might already be fetched, eg.

```py
for node in self.node_set.all():  # could have been populated already -- caller in charge
    for spatial_view in node.spatialview_set.all():
```

Double loop not so great, but that feels like a sign this could be implemented in `Node.xxx()` instead of `Graph.save()`. What do you think?

Then the geometry node is in charge of shuffling the geometry node around. `restore_state_from_serialized_graph()` could call that `Node` method and know what to do without storing state on `Graph`.